src/server/main_service.cc

```diff
@@ -2508,7 +2508,14 @@ GlobalState Service::SwitchState(GlobalState from, GlobalState to) {
   VLOG(1) << "Switching state from " << from << " to " << to;
   global_state_ = to;
 
-  pp_.Await([&](ProactorBase*) { ServerState::tlocal()->set_gstate(to); });
+  pp_.Await([&](ProactorBase*) {
+    ServerState::tlocal()->set_gstate(to);
+    auto* es = EngineShard::tlocal();
+    if (es && to == GlobalState::ACTIVE) {
+      DbSlice& db = namespaces->GetDefaultNamespace().GetDbSlice(es->shard_id());
+      DCHECK(db.IsLoadInProgressZeroInCacheMode());
```

Comment:
I think it will be more clear if you call this function 
VerifyLoadComplete or IsLoadReferenceCleared
and only check that the counter is 0 in this function regardless of cache mode 
This way we will verify the ref count logic not only in cache mode tests

Reply:
Wow what a catch with this dcheck.

Reply:
@adiholden I appreciate you :heart: